zcash_client_sqlite/src/wallet.rs

```diff
@@ -1245,10 +1253,17 @@ pub(crate) fn get_wallet_summary<P: consensus::Parameters>(
         let mut stmt_transparent_balances = tx.prepare(
             "SELECT u.received_by_account_id, SUM(u.value_zat)
              FROM utxos u
-             LEFT OUTER JOIN transactions tx
-             ON tx.id_tx = u.spent_in_tx
              WHERE u.height <= :max_height
-             AND (u.spent_in_tx IS NULL OR (tx.block IS NULL AND tx.expiry_height <= :stable_height))
+             -- and the received txo is unspent
+             AND u.id NOT IN (
+               SELECT transparent_received_output_id
+               FROM transparent_received_output_spends txo_spends
+               JOIN transactions tx
+                 ON tx.id_tx = txo_spends.transaction_id
+               WHERE tx.block IS NOT NULL -- the spending tx is mined
+               OR tx.expiry_height IS NULL -- the spending tx will not expire
+               OR tx.expiry_height > :stable_height -- the spending tx is unexpired
+             )
```

Comment:
Can some of the duplication in the SQL be avoided by defining a view (or views for Sapling and Orchard) of notes that are considered spent? Non-blocking.

Reply:
Oh, the PR does create such views. I don't understand why there is so much SQL duplication, then. It looks pretty error-prone to maintain :-/

Reply:
I want to make a pass after 1.0 is out to do the refactoring to eliminate a whole bunch of the duplication, but I want that to be a pure refactoring rather than part of fixing bugs.